core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/SetClause.kt

```diff
@@ -2,40 +2,51 @@ package ch.ergon.dope.resolvable.clause.model
 
 import ch.ergon.dope.DopeQuery
 import ch.ergon.dope.resolvable.clause.IUpdateSetClause
-import ch.ergon.dope.resolvable.clause.IUpdateUseKeysClause
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.Field
 import ch.ergon.dope.resolvable.formatToQueryString
 import ch.ergon.dope.validtype.ValidType
 
-class SetClause(
-    private val fieldAssignment: Pair<Field<out ValidType>, TypeExpression<out ValidType>>,
-    private vararg val fieldAssignments: Pair<Field<out ValidType>, TypeExpression<out ValidType>>,
-    private val parentClause: IUpdateUseKeysClause,
+class SetClause<T : ValidType>(
```

Comment:
Instead of splitting the first _SetAddignment_ into two arguments I would just pass a SetAssignment as well:
```
class SetClause(
    private val setAssignment: SetAssignment<out ValidType>,
    private vararg val setAssignments: SetAssignment<out ValidType>,
    private val parentClause: IUpdateSetClause,
) : IUpdateSetClause {
```

Reply:
The generic type T is not needed (I think)

Reply:
Yes, I removed the generic type T and am now passing a `SetAssignment`, done